ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/RangeActionFilter.java

```diff
@@ -158,19 +197,61 @@ private void removeAppliedRangeActions(Set<RangeAction> rangeActions) {
     /**
      * Removes from a set of RangeActions the ones with the biggest priority or sensitivity on the most limiting element
      *
-     * @param rangeActions the set of RangeActions to filter
-     * @param numberToRemove       the number of RangeActions to remove from the set
+     * @param rangeActions   the set of RangeActions to filter
+     * @param numberToRemove the number of RangeActions to remove from the set
      */
     private void removeRangeActionsWithBiggestImpact(Set<RangeAction> rangeActions, int numberToRemove) {
-        if (numberToRemove < 0) {
+        if (numberToRemove <= 0) {
             // Nothing to do
         } else if (numberToRemove >= rangeActions.size()) {
             rangeActions.clear();
         } else {
-            List<RangeAction> rangeActionsSortedBySensitivity = rangeActions.stream()
+            removeRangeActionsBySensitivityAndGroupId(rangeActions, numberToRemove);
+        }
+    }
+
+    private void removeRangeActionsBySensitivityAndGroupId(Set<RangeAction> rangeActions, int numberToRemove) {
```

Comment:
I think you don't need to separate this method, since removeRangeActionsWithBiggestImpact became very small